src/parser.ts

```diff
@@ -662,6 +672,165 @@ const TRAVERSE_BUILDER: MarkVisitor<(rhs: TraversalResult | null) => TraversalRe
   },
 }
 
+const SELECTOR_BUILDER: MarkVisitor<null> = {
```

Comment:
Is it worth splitting these `*_BUILDERS` into separate files? This file is getting a little verbose.